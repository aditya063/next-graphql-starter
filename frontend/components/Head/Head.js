import PropTypes from 'prop-types';
import NextHead from 'next/head';

const Head = props => (
  <NextHead>
    <title>{props.title ? `Starter | ${props.title}` : 'Starter'}</title>
  </NextHead>
);

Head.propTypes = { title: PropTypes.string.isRequired };

export default React.memo(Head);
