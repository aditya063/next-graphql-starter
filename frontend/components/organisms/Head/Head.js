import PropTypes from 'prop-types';
import NextHead from 'next/head';
import { title } from '../../../constants';

const Head = props => (
  <NextHead>
    <title>{props.title ? `${title} | ${props.title}` : `${title}`}</title>
  </NextHead>
);

Head.propTypes = { title: PropTypes.string };

export default React.memo(Head);
