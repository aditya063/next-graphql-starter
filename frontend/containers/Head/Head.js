import PropTypes from 'prop-types';
import NextHead from 'next/head';

const Head = React.memo(props => (
  <NextHead>
    <title>{props.title ? `Starter | ${props.title}` : 'Starter'}</title>
  </NextHead>
));

Head.propTypes = {
  title: PropTypes.string.isRequired
};

export default Head;
