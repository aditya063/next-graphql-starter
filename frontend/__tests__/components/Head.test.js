import React from 'react';
import { shallow } from 'enzyme';
import Head from '../../components/Head/Head';

describe('Head', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = { title: 'test' };
    wrapper = shallow(<Head {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    const wrapSnap = wrapper.find({ snapshot: 'Head' });

    expect(wrapSnap).toMatchSnapshot();
  });
});
