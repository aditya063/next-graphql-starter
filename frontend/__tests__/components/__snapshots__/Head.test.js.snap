// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head matches snapshot 1`] = `
<Head>
  <title>
    Starter | test
  </title>
</Head>
`;
